graphql/resolvers.go

```diff
@@ -2530,6 +2510,127 @@ func (r *queryResolver) BbGetCreatedTickets(ctx context.Context, taskID string)
+		}
+
+		// If we try to access a version that does not have the Activated flag we must manually verify it
+		// After we verify if a version is activated we cache that field so subsequent queries are faster.
+		if v.Activated == nil {
+			defaultSort := []task.TasksSortOrder{
+				{Key: task.DisplayNameKey, Order: 1},
+			}
+			opts := data.TaskFilterOptions{
+				Sorts: defaultSort,
+			}
+			tasks, _, err := r.sc.FindTasksByVersion(v.Id, opts)
+			if err != nil {
+				return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error fetching tasks for version %s : %s", v.Id, err.Error()))
+			}
+			if !task.AnyActiveTasks(tasks) {
+				err = v.SetNotActivated()
+				if err != nil {
+					return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error updating version activated status for %s, %s", v.Id, err.Error()))
+				}
+			} else {
+				err = v.SetActivated()
+				if err != nil {
+					return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error updating version activated status for %s, %s", v.Id, err.Error()))
+				}
+			}
+		}
+		mainlineCommitVersion := MainlineCommitVersion{}
+
+		// If a version is activated we append it directly to our returned list of mainlineCommits
+		// If a version is not activated we roll up all the unactivated versions that are sequentially near each other
+		// into a single MainlineCommitVersion and then append it to our returned list
+		if utility.FromBoolPtr(v.Activated) {
+			activatedVersionCount++
+			mainlineCommits.NextPageOrderNumber = &v.RevisionOrderNumber
+			mainlineCommitVersion.Version = &apiVersion
+
+		} else {
+			// If we have any versions already we should check the most recent one first otherwise create a new one
+			if len(mainlineCommits.Versions) > 0 {
+				lastMainlineCommit := mainlineCommits.Versions[len(mainlineCommits.Versions)-1]
+
+				// If the previous mainlineCommit contains rolled up unactivated versions append the latest RolledUp unactivated version
+				if lastMainlineCommit.RolledUpVersions != nil {
+					lastMainlineCommit.RolledUpVersions = append(lastMainlineCommit.RolledUpVersions, &apiVersion)
+				} else {
+					mainlineCommitVersion.RolledUpVersions = []*restModel.APIVersion{&apiVersion}
+				}
+
+			} else {
+				mainlineCommitVersion.RolledUpVersions = []*restModel.APIVersion{&apiVersion}
+
+			}
+
+		}
+
+		// Only add a mainlineCommit if a new one was added and its not a modified existing RolledUpVersion
+		if mainlineCommitVersion.Version != nil || mainlineCommitVersion.RolledUpVersions != nil {
+			mainlineCommits.Versions = append(mainlineCommits.Versions, &mainlineCommitVersion)
+		}
+	}
+
+	return &mainlineCommits, nil
+}
+
+type versionResolver struct{ *Resolver }
+
+// Returns grouped build variants for a version. Will not return build variants for unactivated versions
+func (r *versionResolver) BuildVariants(ctx context.Context, v *restModel.APIVersion, options *BuildVariantOptions) ([]*GroupedBuildVariant, error) {
+	if !utility.FromBoolPtr(v.Activated) {
```

Comment:
Do we need to consider if v.Activated is nil, or would the MainlineCommits resolver have been guaranteed to hit first?

Reply:
In this case it is guaranteed to have been hit first. But if we want to use this for another field that returns a version it is not guaranteed. There will be a super low likely hood of it being nil unless someone decides to view a really old version that was introduced before this field was introduced

Reply:
I added handling for that edge case just in case.